### PR TITLE
feat: default event tracking for Swift SDK

### PR DESCRIPTION
**Pull request by justin-fiedler**
## Description

[AMP-82700](https://amplitude.atlassian.net/browse/AMP-82700)

* https://pr-982.d19s7xzcva2mw3.amplifyapp.com/data/sdks/ios-swift/#tracking-default-events
* default event tracking for Swift SDK
* Mostly copied from [maintenance iOS docs](https://pr-982.d19s7xzcva2mw3.amplifyapp.com/data/sdks/ios/#tracking-default-events), code examples adjusted for Swift SDK

[AMP-82700]: https://amplitude.atlassian.net/browse/AMP-82700?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/data/sdks/ios-swift/index.md [hunk 1]
```diff
@@ -106,6 +106,160 @@ identify.set(property: "color", value: "green")
 amplitude.identify(identify: identify)
 ```
 
+### Tracking default events
+
+Starting from release v0.6.0, the SDK is able to track more default events. It can be configured to track the following events automatically:
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:111]:
```suggestion
Starting from release v0.6.0, the SDK can track more default events. You can configure it to track the following events by default:
```

File: docs/data/sdks/ios-swift/index.md [hunk 2]
```diff
@@ -106,6 +106,160 @@ identify.set(property: "color", value: "green")
...
+
+- Sessions [^1]
+- App lifecycles
+- Screen views[^2]
+- Deep links [^3]
+
+[^1]:
+Session tracking is the same as supported in previous versions, which was previously enabled/disabled via the [`trackingSessionEvents`](#configuration) configuration.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:119]:
```suggestion
Session tracking is the same as supported in previous versions, which was previously enabled/disabled with the [`trackingSessionEvents`](#configuration) configuration.
```

File: docs/data/sdks/ios-swift/index.md [hunk 3]
```diff
@@ -106,6 +106,160 @@ identify.set(property: "color", value: "green")
...
+[^2]:
+Screen views are only supported in UIKit. For Swift UI, you will need to call the corresponding methods manually.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:121]:
```suggestion
Screen views are supported in UIKit. For Swift UI, you need to manually call the corresponding methods.
```

File: docs/data/sdks/ios-swift/index.md [hunk 4]
```diff
@@ -106,6 +106,160 @@ identify.set(property: "color", value: "green")
...
+[^3]:
+Deep link tracking is not fully automated, you will need to call the corresponding methods manually.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:123]:
```suggestion
Deep link tracking is not fully automated. You need to manually call the corresponding methods.
```

File: docs/data/sdks/ios-swift/index.md [hunk 5]
```diff
@@ -106,6 +106,160 @@ identify.set(property: "color", value: "green")
...
+`defaultTracking.appLifecycles` | Optional. `boolean` | `false` | Enables application lifecycle events tracking. If value is `true`, Amplitude tracks application installed, application updated, application opened, and application backgrounded events.<br /><br />Event properties tracked includes: `[Amplitude] Version`,<br /> `[Amplitude] Build`,<br /> `[Amplitude] Previous Version`, `[Amplitude] Previous Build`, `[Amplitude] From Background`<br /><br />See [Tracking application lifecycles](#tracking-application-lifecycles) for more information.|
+`defaultTracking.screenViews` | Optional. `boolean` | `false` | Enables screen views tracking. If value is `true`, Amplitude tracks screen viewed events.<br /><br />Event properties tracked includes: `[Amplitude] Screen Name`<br /><br />See [Tracking screen views](#tracking-screen-views) for more information.|
+
+You can enable Amplitude to start tracking all events mentioned above, use the code sample below.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:132]:
```suggestion
Use the code sample below to enable the above-mentioned tracking events.
```

**Review by yuhao900914 — APPROVED** [review]

File: docs/data/sdks/ios-swift/index.md [hunk 6]
```diff
@@ -106,6 +106,160 @@ identify.set(property: "color", value: "green")
...
+
+    ```swift
+    var instance = Amplitude(configuration: Configuration(
+        apiKey: "API_KEY",
+        defaultTracking: DefaultTrackingOptions.ALL
+    ))
+    ```
+
+!!!note
+Amplitude may add more events in a future version, and this configuration enables tracking for those events as well.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md]:
```suggestion
This configuration enables tracking for any events Amplitude may add in future releases.
```

File: docs/data/sdks/ios-swift/index.md [hunk 7]
```diff
@@ -106,6 +106,160 @@ identify.set(property: "color", value: "green")
...
+
+Similarly, you can disable Amplitude to track all events mentioned above with the code sample below.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:147]:
```suggestion
Use the code sample below to disable tracking the above-mentioned tracking events.
```

File: docs/data/sdks/ios-swift/index.md [hunk 8]
```diff
@@ -106,6 +106,160 @@ identify.set(property: "color", value: "green")
...
+=== "Swift"
+
+    ```swift
+    var instance = Amplitude(configuration: Configuration(
+        apiKey: "API_KEY",
+        defaultTracking: DefaultTrackingOptions.NONE
+    ))
+    ```
+
+You can also customize the tracking with `DefaultTrackingOptions`, see code sample below.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:158]:
```suggestion
Customize the tracking with `DefaultTrackingOptions`.
```

File: docs/data/sdks/ios-swift/index.md [hunk 9]
```diff
@@ -106,6 +106,162 @@ identify.set(property: "color", value: "green")
...
+            sessions: true,
+            appLifecycles: false,
+            screenViews: false
+        )
+    ))
+    ```
+
+#### Tracking sessions
+
+Session tracking is enabled by default. You can enable Amplitude to start tracking session events by setting `defaultTracking.sessions` to `true`. Refer to the code sample below.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:175]:
```suggestion
Amplitude enables session tracking by default. Set `defaultTracking.sessions` to `true` to  track session events.
```

File: docs/data/sdks/ios-swift/index.md [hunk 10]
```diff
@@ -106,6 +106,162 @@ identify.set(property: "color", value: "green")
...
+
+For more information about session tracking, refer to [User sessions](#user-sessions).
+
+!!!note
+
+    `trackingSessionEvents` is deprecated and replaced with `defaultTracking.sessions`.
+
+#### Tracking application lifecycles
+
+You can enable Amplitude to start tracking application lifecycle events by setting `defaultTracking.appLifecycles` to `true`. Refer to the code sample below.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:198]:
```suggestion
Set `defaultTracking.appLifecycles` to `true` to enable Amplitude to track application lifecycle events.
```

File: docs/data/sdks/ios-swift/index.md [hunk 11]
```diff
@@ -106,6 +106,162 @@ identify.set(property: "color", value: "green")
...
+        apiKey: "API_KEY",
+        defaultTracking: DefaultTrackingOptions(
+            sessions: false,
+            appLifecycles: true,
+            screenViews: false
+        )
+    ))
+    ```
+
+After enabling this setting, Amplitude will track the following events:
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:213]:
```suggestion
When you enable this setting, Amplitude tracks the following events:
```

File: docs/data/sdks/ios-swift/index.md [hunk 12]
```diff
@@ -106,6 +106,162 @@ identify.set(property: "color", value: "green")
...
+
+-`[Amplitude] Application Installed`, this event fires when a user opens the application for the first time right after installation, by observing the `UIApplicationDidFinishLaunchingNotification` notification underneath.
+-`[Amplitude] Application Updated`, this event fires when a user opens the application after updating the application, by observing the `UIApplicationDidFinishLaunchingNotification` notification underneath.
+-`[Amplitude] Application Opened`, this event fires when a user launches or foregrounds the application after the first open, by observing the `UIApplicationDidFinishLaunchingNotification` or `UIApplicationWillEnterForegroundNotification` notification underneath.
+-`[Amplitude] Application Backgrounded`, this event fires when a user backgrounds the application, by observing the `UIApplicationDidEnterBackgroundNotification` notification underneath.
+
+#### Tracking screen views
+
+You can enable Amplitude to start tracking screen view events by setting `defaultTracking.screenViews` to `true`. Refer to the code sample below.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:222]:
```suggestion
Set `defaultTracking.screenViews` to `true` to enable Amplitude to track screen view events.
```

File: docs/data/sdks/ios-swift/index.md [hunk 13]
```diff
@@ -106,6 +106,162 @@ identify.set(property: "color", value: "green")
...
+
+!!!warning
+    This feature is only supported in UIKit. For Swift UI, you will need to track the corresponding event manually instead.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:225]:
```suggestion
    This feature is supported in UIKit. For Swift UI, track the corresponding event manually.
```

File: docs/data/sdks/ios-swift/index.md [hunk 14]
```diff
@@ -106,6 +106,162 @@ identify.set(property: "color", value: "green")
...
+            screenViews: true
+        )
+    ))
+
+    // Swift UI
+    instance.configuration.defaultTracking.screenViews = false
+    instance.track(ScreenViewedEvent(screenName: "Screen Name"))
+    ```
+
+After enabling this setting, Amplitude will track the `[Amplitude] Screen Viewed` event with the screen name property. This property value is read from the controller class metadata with `viewDidAppear` method swizzling.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:245]:
```suggestion
When you enable this setting, Amplitude tracks the `[Amplitude] Screen Viewed` event with the screen name property. Amplitude reads this value from the controller class metadata `viewDidAppear` method swizzling.
```

File: docs/data/sdks/ios-swift/index.md [hunk 15]
```diff
@@ -106,6 +106,162 @@ identify.set(property: "color", value: "green")
...
+
+#### Tracking deep links
+
+Deeplinks are not automated at this time. You will need to track the corresponding events manually instead.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:249]:
```suggestion
Deeplink tracking is not automated. To track deeplinks, track the corresponding events.
```

File: docs/data/sdks/ios-swift/index.md [hunk 16]
```diff
@@ -106,6 +106,162 @@ identify.set(property: "color", value: "green")
...
+    var instance = Amplitude(configuration: Configuration(
+        apiKey: "API_KEY"
+    ))
+
+    instance.track(DeepLinkOpenedEvent(url: URL()))
+    instance.track(DeepLinkOpenedEvent(url: "url", referrer:"refferer"))
+    instance.track(DeepLinkOpenedEvent(activity: activity))
+    ```
+
+Amplitude will track the `[Amplitude] Deep Link Opened` event with the URL and referrer information.
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/index.md:263]:
```suggestion
Amplitude  tracks the `[Amplitude] Deep Link Opened` event with the URL and referrer information.
```

**Review by markzegarelli — CHANGES_REQUESTED** [review]
@justin-fiedler Added some comments to make your additions active-voice and present tense.

I will take a separate pass through the article to make sure any existing content matches this style after this is merged.

**Comment by justin-fiedler** [comment]
Thanks @markzegarelli! Reads much cleaner now!

**Review by markzegarelli — APPROVED** [review]

**Pull request merged**